src/material/slider/slider.ts

```diff
@@ -488,9 +489,13 @@ export class MatSlider extends _MatSliderMixinBase
   /** Keeps track of the last pointer event that was captured by the slider. */
   private _lastPointerEvent: MouseEvent | TouchEvent | null;
 
+  /** Used to subscribe to global move and end events */
+  protected _document?: Document;
+
   constructor(elementRef: ElementRef,
               private _focusMonitor: FocusMonitor,
               private _changeDetectorRef: ChangeDetectorRef,
+              @Inject(DOCUMENT) document: any,
```

Comment:
Can you move this to be the last argument and make it optional? Technically changing the constructor is a breaking API change

Reply:
```ts
...
/** @breaking-change 11.0.0 make document required */
@Optional @Inject(DOCUMENT) document?: any,
...
```

Reply:
Change made.